src/program/lwaftr/query/query.lua

```diff
@@ -87,17 +93,52 @@ local function print_counters (tree, filter)
    end
 end
 
+-- Return the pid that was specified, unless it was a leader process,
+-- in which case, return the follower pid that actually has useful counters.
+local function pid_to_parent(pid)
+   -- It's meaningless to get the parent of a nil 'pid'.
+   if not pid then return pid end
+   local pid = tonumber(pid)
+   for _, name in ipairs(shm.children("/")) do
+      local p = tonumber(name)
+      if p and ps.is_worker(p) then
+         local leader_pid = tonumber(ps.get_leader_pid(p))
+         -- If the precomputed by-name pid is the leader pid, set the pid
+         -- to be the follower's pid instead to get meaningful counters.
+      if leader_pid == pid then pid = p end
+      end
+   end
+   return pid
+end
+
 function run (raw_args)
    local opts, pid, counter_name = parse_args(raw_args)
-   if tostring(pid) and not counter_name then
-      counter_name, pid = pid, nil
-   end
+   if tostring(pid) then pid = pid_to_parent(pid) end
```

Comment:
This code runs most of the times, but when "lwaftr query --name <name> <counter>", then pid is actually the counter name, and counter_name is nil. Now querying by name and counter is working because it always returns all values. The fix is not correct.

I think we should distinguish between two cases, that are exclusive querying by pid or querying by name. Maybe renamind pid and counter_name as arg1 and arg2 and properly assigning their values to pid and counter_name depending on each case would make things clearer.